05-UI-ReactJS/react-builder-master/source/list.js

```diff
@@ -10,30 +11,32 @@ class List extends React.Component {
  render() {
    return (
      <ul>
-       {this.renderItems()}
+       {this.renderItems() }
      </ul>
    );
  }
 
+ editMovie(index){
+    let mo = JSON.parse(localStorage.getItem("movies"))
+    console.log(index);
+    console.log(mo[index].title);
+}
+
  renderItems () {
-   let favourites = this.props.movies.filter( function(item) {
-     return item.fav;
-   })
-   console.log(favourites);
-   return favourites.map(this.renderItem);
- }
+    if (localStorage.getItem("movies") !== null){
```

Comment:
you should validate with the array movies in *this.props* to desacoplate the component of localStorage (think that this way, if you want to implement a database, you will have to refactor the components)